File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        return new int[]{gdc(n, m), lcm(n, m)};
    }

    public static int gdc(int n, int m) {
        if (n < m) {
            int tmp = n;
            n = m;
            m = tmp;
        }
        while (m != 0) {
            int r = n % m;
            n = m;
            m = r;
        }
        return n;
    }

    public static int lcm(int n, int m) {
        return n * m / gdc(n, m);
    }
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.swift
func solution(_ n:Int, _ m:Int) -> [Int] {
    func gcd(n: Int, m: Int) -> Int {
        var n = n
        var m = m
        if (n < m) {
            var tmp = n;
            n = m;
            m = tmp;
        }
        while (m != 0) {
            var r = n % m;
            n = m;
            m = r;
        }
        return n;
    }

    func lcm(n: Int, m: Int) -> Int {
        return n * m / gcd(n: n, m: m);
    }
    
    return [gcd(n: n, m: m), lcm(n: n, m: m)]
}